src/repositories/donationRepository.ts

```diff
@@ -72,3 +72,23 @@ export const findDonationById = async (
     .leftJoinAndSelect('donation.project', 'project')
     .getOne();
 };
+
+export const findUsersWhoDonatedToProject = async (
+  projectId: number,
+): Promise<{ walletAddress: string; email?: string }[]> => {
+  const donations = await Donation.createQueryBuilder('donation')
+    .leftJoin('donation.user', 'user')
+    .addSelect(['user.walletAddress', 'user.email'])
+    .distinctOn(['user.walletAddress'])
+    .where(`"projectId"=:projectId`, {
+      projectId,
+    })
+    .getMany();
+
+  return donations.map(donation => {
+    return {
+      walletAddress: donation.user.walletAddress?.toLowerCase() as string,
```

Comment:
You can avoid this loop entirely by using ```.getRawMany()``` instead, this returns the selected data only as json and not entity mapped data.
And in the select you can do directly:
```
.select('LOWER(user.walletAddress) AS "walletAddress", user.email as email')
```

This should return:
```
[{ walletAddress: 'xxxxx', email: 'xxxx' }]
```

Reply:
From what I see in the code we have 2 loops, 
Fetch query -> map data correctly -> map to enqueue.

If we delete the middle map this would greatly decrease code complexity.

Reply:
Done